codex/stores/networkstore.nim

```diff
@@ -37,21 +37,18 @@ method getBlock*(self: NetworkStore, cid: Cid): Future[?!bt.Block] {.async.} =
 
   trace "Getting block from local store or network", cid
 
-  let
-    getRes = await self.localStore.getBlock(cid)
-
-  if getRes.isErr:
-    if not (getRes.error of BlockNotFoundError): return getRes
+  without blk =? await self.localStore.getBlock(cid), error:
+    if not (error of BlockNotFoundError): return failure error
     trace "Block not in local store", cid
-    try:
+    without blk =? (await self.engine.requestBlock(cid)).catch, error:
```

Comment:
I think we can just return the result from `requestBlock` directly, all we're doing is passing converting a `Result` to another `Result`?

Reply:
We can. It seemed a little more linear to not do that, e.g. with respect to the `# TODO comments` which this PR inherited, i.e. leaving room for refinement of the logic there.

Reply:
Though... maybe there isn't too much room for refinement in that case, at present.

Reply:
Change made.